testsuite/standalone/src/test/java/org/jboss/as/test/integration/management/cli/CliCompletionTestCase.java

```diff
@@ -118,6 +118,36 @@ public void complexOperationWithObjectAsAttributeTest() {
             assertEquals(candidates.toString(), Arrays.asList(","), candidates);
         }
     }
+    /**
```

Comment:
Nice to add some tests. It would be great to cover the fix you are putting in place. 
Something like:
CommandContext ctx = CLITestUtil.getCommandContext(TestSuiteEnvironment.getServerAddress(),          TestSuiteEnvironment.getServerPort(), System.in, System.out);
ctx.connectController();
// cd into a type to validate that completion tha toccurs in this context is correct.
ctx.handle("cd /system-property");
try {
            {
                String cmd = "read-attribute ";
                List<String> candidates = new ArrayList<>();
                ctx.getDefaultCommandCompleter().complete(ctx,
                        cmd, cmd.length(), candidates);
                candidates = complete(ctx, cmd, null);
            }
} finally {
  ctx.terminateSession();
}

Reply:
Thanks for the suggestion @jfdenise. Please feel free to let me know if any further changes needed.